homeassistant/components/steam_online/config_flow.py

```diff
@@ -0,0 +1,222 @@
+        """Confirm reauth dialog."""
+        if user_input is not None:
+            return await self.async_step_user()
+
+        self._set_confirm_only()
+        return self.async_show_form(step_id="reauth_confirm")
+
+
+class SteamOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle Steam client options."""
+
+    def __init__(self, entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.entry = entry
+        self.options = dict(entry.options)
+
+    async def async_step_init(
+        self, user_input: dict[str, dict[str, str]] | None = None
+    ) -> FlowResult:
+        """Manage Steam options."""
+        if user_input is None:
+            try:
+                interface = await self.hass.async_add_executor_job(
+                    steam.api.interface, "ISteamUser"
+                )
+                friends = await self.hass.async_add_executor_job(
+                    ft.partial(
+                        interface.GetFriendList, steamid=self.entry.data[CONF_ACCOUNT]
+                    )
+                )
+                friends = (
+                    await self.hass.async_add_executor_job(friends.get, "friendslist")
+                )["friends"]
+                _users_str = str([user["steamid"] for user in friends])
+                names = await self.hass.async_add_executor_job(
+                    ft.partial(interface.GetPlayerSummaries, steamids=_users_str)
+                )
+                names = (await self.hass.async_add_executor_job(names.get, "response"))[
+                    "players"
+                ]["player"]
+                users = {
+                    name["steamid"]: {"name": name["personaname"], "enabled": False}
+                    for name in names
+                }
+            except steam.api.HTTPTimeoutError:
+                users = self.options[CONF_ACCOUNTS]
+            _users = users | self.options[CONF_ACCOUNTS]
+            self.options[CONF_ACCOUNTS] = {
+                k: v
+                for k, v in _users.items()
+                if k in users.keys() or self.options[CONF_ACCOUNTS][k]["enabled"]
+            }
+
+        else:
+            await self.hass.config_entries.async_unload(self.entry.entry_id)
+            for k in self.options[CONF_ACCOUNTS].keys():
+                if (
+                    self.options[CONF_ACCOUNTS][k]["enabled"]
+                    and k not in user_input[CONF_ACCOUNTS]
+                    and (
+                        entity_id := er.async_get(self.hass).async_get_entity_id(
+                            Platform.SENSOR, DOMAIN, f"sensor.steam_{k}"
+                        )
+                    )
+                ):
+                    er.async_get(self.hass).async_remove(entity_id)
+            channel_data = {
+                CONF_ACCOUNTS: {
+                    k: {
+                        "name": v["name"],
+                        "enabled": bool(k in user_input[CONF_ACCOUNTS]),
```

Comment:
this is already a boolean. 
```suggestion
                        "enabled": k in user_input[CONF_ACCOUNTS],
```